HOWTO.rst

```diff
@@ -4103,22 +4103,24 @@ Measurements and reporting
 
 .. option:: log_unix_epoch=bool
 
-	If set, fio will log Unix timestamps to the log files produced by enabling
-	write_type_log for each log type, instead of the default zero-based
-	timestamps.
+    Backwards compatible alias for log_alternate_epoch.
 
 .. option:: log_alternate_epoch=bool
 
-	If set, fio will log timestamps based on the epoch used by the clock specified
-	in the log_alternate_epoch_clock_id option, to the log files produced by
-	enabling write_type_log for each log type, instead of the default zero-based
-	timestamps.
+    If set, fio will log timestamps based on the epoch used by the clock
+    specified in the clock_id option, to the log files produced by enabling
+    write_type_log for each log type, instead of the default zero-based
+    timestamps.
 
 .. option:: log_alternate_epoch_clock_id=int
 
-	Specifies the clock_id to be used by clock_gettime to obtain the alternate epoch
-	if either log_unix_epoch or log_alternate_epoch are true. Otherwise has no
-	effect. Default value is 0, or CLOCK_REALTIME.
+    Backwards compatible alias for clock_id.
+
+.. option:: clock_id=int
```

Comment:
The name `clock_id` is not specific enough and risks being confused with the option `clocksource`. How about something like `log_clock_id`?

Reply:
Gotcha. It used to be `log_alternate_epoch_clock_id`. The only problem with `log_clock_id` is that it now affects the job start time recorded in the json, which is independent of the `log_alternate_epoch` setting which controls if the log files use times based on this (needs better name) `clock_id`.

How about `alternate_epoch_clock_id`, and then I rename the `clock_gettime_job_start` variable and json entry to `alternate_epoch_job_start` ? I think that may work pretty well if it sounds ok to you.

Reply:
I'm happy to do that (completely separate option and `struct thread_data` member). I started out planning to do that but then ended up the way I submitted because, well, the options are more related than they seem from the documentation side. (You may already be aware of this, but just in case.) Here's the background.

Before I started using fio, there was a `log_unix_epoch` option. The docs say it logs Unix timestamps, and this is *kindof* true. What it actually does is remember the Unix timestamp when the job started (obtained via `clock_gettime`) and then add the elapsed job time (obtained based on the configured `clocksource`) to obtain a Unix-format timestamp for each entry. This is actually different than what you'd get by calling `clock_gettime` to get the Unix timestamp for each I/O operation, since Unix time is not monotonic and is affected by NTP.

Then I added `log_alternate_epoch`, which is just like `log_unix_epoch` but lets you use any `clock_id` instead of `CLOCK_REALTIME` (used for `log_unix_epoch`). (Interesting aside - in the case of using `CLOCK_MONOTONIC_RAW`, it turns out that the job start time plus the elapsed job time *can* actually be the same as what you'd get calling `clock_gettime` for each I/O,)

Anyways, that's why job start time ends up so "tied" to the options for log file timestamps. But there is no reason the log file timestamps and the recorded job start time have to share the same clock_id.

If I had a time machine, I'd stop any of the `log` options from being created, and the log files would always record time elapsed since job started, and a single `job_start_time` would be the only mechanism for recording time using another `clock_id`, since that is what is really happening in the implementation. The "alternate" timestamps in the log files are really an illusion, if that makes sense.

But I know that can't happen. Maybe all of this is really just another good argument for what you suggested, as it keeps the (imho) log timestamps and recording job start time completely separate in terms of implementation, so either one could be revamped in the future without having to worry about the other.

Just let me know what you think is best (just make it two separate things or try to come up with an option name for both purposes); I'm happy to do it either way.

Thanks for listening to my rambling thoughts. I

Reply:
Thanks for the recap. I think it would be cleaner to just have an entirely separate start time feature.

Reply:
Cool, I'll do that then. Thanks for the feedback!